js/boaexchange.js

```diff
@@ -0,0 +1,1054 @@
+        return this.parseTradingLimits (this.safeValue (response, 'data', {}));
+    }
+
+    async fetchTransactions (symbol = undefined, since = 0, limit = 0, params = {}) {
+        await this.loadMarkets ();
+        let response = await this.privateGetTransactions ();
+        return this.parseTransactions (response['data']);
+    }
+
+    async fetchWithdrawals (code = undefined, since = 0, limit = 0, params = {}) {
+        await this.loadMarkets ();
+        const request = { 'limit': limit, 'page': since };
+        let currency = undefined;
+        if (code !== undefined) {
+            currency = this.currency (code);
+            request['coin'] = currency['id'];
+        }
+        const response = await this.privateGetWithdraws (this.extend (request, params));
+        return this.parseTransactions (response['data'], currency);
+    }
+
+    async transfer (code, amount, accountFrom = undefined, accountTo = undefined, params = {}) {
+        await this.loadMarkets ();
+        let currency = this.currency (code);
+        let request = {
+            'to': accountTo,
+            'coin': currency.name,
+            'from': accountFrom,
+            'amount': this.parseFloat (amount),
+        };
+        let response = await this.privateGeAccountTransfer (this.extend (request, params));
+        return {
+            'info': response,
+            'id': undefined,
+        };
+    }
+
+    async withdraw (code, amount, address, tag = undefined, params = {}) {
+        this.checkAddress (address);
+        await this.loadMarkets ();
+        let currency = this.currency (code);
+        let request = {
+            'label': currency['id'],
+            'amount': amount,
+            'address': address,
+            'password': params['password'],
+        };
+        let response = await this.privateGetAccountWithdraw (this.extend (request, params));
+        let id = undefined;
+        if ('data' in response) {
+            if ('withdraw' in response['data'])
+                id = response['data']['withdraw']['id'];
+        }
+        return {
+            'info': response,
+            'id': id,
+        };
+    }
+
+    orderToTrade (order) {
+        // this entire method should be moved to the base class
+        const timestamp = this.safeInteger (order, 'created');
+        const market = this.marketId (this.safeString (order, 'market'));
+        return {
+            'id': this.safeString (order, 'id'),
+            'side': this.safeString (order, 'side'),
+            'order': this.safeString (order, 'id'),
+            'price': this.safeFloat (order, 'price'),
+            'amount': this.safeFloat (order, 'amount'),
+            'cost': this.safeFloat (order, 'amount_coin'),
+            'symbol': market['symbol'],
+            'timestamp': timestamp,
+            'datetime': this.iso8601 (timestamp),
+            'fee': undefined,
+            'info': order,
+        };
+    }
+
+    ordersToTrades (orders) {
```

Comment:
Looks like methods `ordersToTrades` and `orderToTrades` are not used anywhere in this implementation, so we can probably drop them as well.

Reply:
done